File: app/src/main/java/com/yuk/miuiHomeR/ui/base/BasePreferenceFragment.java
package com.yuk.miuiHomeR.ui.base;

import android.content.Context;
import android.os.Bundle;

import com.yuk.miuiHomeR.utils.PrefsUtils;

import java.util.Objects;

import moralnorm.preference.PreferenceFragmentCompat;
import moralnorm.preference.PreferenceManager;

public class BasePreferenceFragment extends PreferenceFragmentCompat {


    public void onCreate(Bundle savedInstanceState, int prefs_default) {
        super.onCreate(savedInstanceState);
        try {
            getPreferenceManager().setSharedPreferencesName(PrefsUtils.mPrefsName);
            getPreferenceManager().setSharedPreferencesMode(Context.MODE_PRIVATE);
            getPreferenceManager().setStorageDeviceProtected();
            PreferenceManager.setDefaultValues(requireActivity(), prefs_default, false);
        } catch (Throwable throwable) {
            throwable.printStackTrace();
        }
    }

    @Override
    public void onCreatePreferences(Bundle bundle, String s) {

    }
}

File: app/src/main/java/com/yuk/miuiHomeR/ui/base/BasePreferenceFragment.kt
package com.yuk.miuiHomeR.ui.base

import android.content.Context
import android.os.Bundle
import com.yuk.miuiHomeR.utils.PrefsUtils
import moralnorm.preference.PreferenceFragmentCompat
import moralnorm.preference.PreferenceManager

open class BasePreferenceFragment : PreferenceFragmentCompat() {

    fun onCreate(savedInstanceState: Bundle?, prefs_default: Int) {
        super.onCreate(savedInstanceState)
        try {
            preferenceManager.sharedPreferencesName = PrefsUtils.mPrefsName
            preferenceManager.sharedPreferencesMode = Context.MODE_PRIVATE
            preferenceManager.setStorageDeviceProtected()
            PreferenceManager.setDefaultValues(requireActivity(), prefs_default, false)
        } catch (throwable: Throwable) {
            throwable.printStackTrace()
        }
    }

    override fun onCreatePreferences(bundle: Bundle?, s: String?) {}
}